include/onnxruntime/core/framework/execution_provider.h

```diff
@@ -278,15 +281,15 @@ class IExecutionProvider {
 
   // helper to generate ids that are unique to model and deterministic, even if the execution provider is shared across
   // multiple sessions.
-  class ModelMetadefIdGenerator {
+  class ModelIdGenerator {
    public:
-    int GenerateId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash);
+    int GenerateId(const onnxruntime::GraphViewer& graph_viewer, uint64_t& model_hash, bool hash_model_name = false, bool hash_nodes = false);
```

Comment:
nit: I know this is an internal function. But still would be good to add a line of documentation for this function stating what is used for hashing if both hash_model_name and hash_nodes are false. Also, it's not evident that path is not used when hash_model_name is true.

Reply:
comments are added